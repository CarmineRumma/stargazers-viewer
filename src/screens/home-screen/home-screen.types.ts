import {TAutocompleteDropdownItem} from 'react-native-autocomplete-dropdown';
import {RouteProp} from '@react-navigation/native';
import {RootStackParamList} from '@stargazers/navigators/navigator.types';
import {NativeStackScreenProps} from '@react-navigation/native-stack';

export interface Autocomplete extends TAutocompleteDropdownItem {}
export interface OwnerSuggestionType extends TAutocompleteDropdownItem {}
