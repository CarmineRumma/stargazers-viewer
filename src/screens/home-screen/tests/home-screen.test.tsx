import React from 'react';

import renderer from 'react-test-renderer';
import {HomeScreen} from '../home-screen';

import {AutocompleteDropdown, AutocompleteDropdownRef} from 'react-native-autocomplete-dropdown';

jest.mock('react-native-vector-icons/Feather', () => 'Feather');
const goBack = jest.fn();
const navigate = jest.fn();
const setOptions = jest.fn();
const addListener = jest.fn();
const mockedNavigation = jest.fn();

jest.mock('react-native-autocomplete-dropdown');
describe('HomeScreen', () => {
});
