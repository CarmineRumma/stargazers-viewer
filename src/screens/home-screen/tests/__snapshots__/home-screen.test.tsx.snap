// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source="test-file-stub"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "padding": 0,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "bottom": 0,
          "fontSize": 12,
          "height": 80,
          "lineHeight": 20,
          "maxWidth": 750,
          "paddingHorizontal": 22,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "space-between",
            "alignItems": "center",
            "flexDirection": "row",
            "fontFamily": "HelveticaNeue-Medium",
            "fontSize": 14,
            "justifyContent": "center",
            "paddingBottom": 10,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#383B42",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 12,
                "lineHeight": 24,
                "textTransform": "uppercase",
              }
            }
          >
            about
          </Text>
        </View>
        <Text
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
        >
          |
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#383B42",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 12,
                "lineHeight": 24,
                "textTransform": "uppercase",
              }
            }
          >
            License
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#383B42",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 13,
          }
        }
      >
        login_footer_txt
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "padding": 0,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.12)",
            "height": "100%",
            "opacity": 0,
            "position": "absolute",
            "width": "100%",
            "zIndex": 1000,
          }
        }
      />
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0,
            "borderColor": "#FFFFFF",
            "borderRadius": 20,
            "borderWidth": 1,
            "elevation": 1,
            "height": 310,
            "maxWidth": 400,
            "opacity": 0,
            "padding": 20,
            "paddingTop": 15,
            "position": "absolute",
            "shadowColor": "#414757",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
            "transform": Array [
              Object {
                "translateY": 180,
              },
            ],
            "width": "85%",
            "zIndex": 2000,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#9867F0",
              "borderRadius": 12,
              "flexDirection": "row",
              "marginTop": 12,
              "padding": 16,
              "width": "100%",
            }
          }
        >
          <Feather
            color="#FFFFFF"
            name="alert-circle"
            size={18}
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 16,
                "marginEnd": 16,
                "marginStart": 8,
              }
            }
          >
            Type a GitHub user below and select from the list to see his repositories.
          </Text>
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#B4B4B4",
              "borderColor": "#505050",
              "borderRadius": 40,
              "minWidth": 150,
              "padding": 10,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#505050",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 18,
                "textAlign": "center",
              }
            }
          >
            Continua
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
